### ISSUE TITLE
About the loss_distill

### ISSUE THREAD

**Issue by sushizixin** [uta-smile/TCL#12]
Hi, thank you for the excellent work and the release of the code!

I am a little confused about the approach to calculating `loss_distill` in _line 1429_ of `xbert.py` as shown in 

                      loss_distill = -torch.sum(F.log_softmax(prediction_scores, dim=1)*soft_labels,dim=-1)

I think the size of both `prediction_scores` and `soft_labels` would be _(batch_size, seq_len, vocab_size)_. And `F.softmax` is used in the last dimension for `soft_labels` in _line 237_ of `model_pretrain.py`, as shown in

                      mlm_output = self.text_encoder(input_ids, 
                                                     attention_mask = text.attention_mask,
                                                     encoder_hidden_states = image_embeds,
                                                     encoder_attention_mask = image_atts,      
                                                     return_dict = True,
                                                     labels = labels,   
                                                     soft_labels = F.softmax(logits_m,dim=-1),
                                                     alpha = alpha
                                                    )
Why is `F.log_softmax` used in the second dimension (dimension of _seq_len_) for `prediction_scores`?

**Comment by viyjy** [comment]
Hi, thanks for your interest in our work.
Have you printed out the size of `prediction_scores` and `soft_labels`?

**Comment by sushizixin** [comment]
Hi, thanks for your response. 
I printed out the size of them by setting `batch_size` in `Pretrain.yaml` to _2_ and find that they have the same size, e.g. _(2, 11, 30522)_.

**Comment by viyjy** [comment]
Thanks for pointing out this bug. I fixed this issue and merged it to the mainline. More details can be found in this commit: https://github.com/uta-smile/TCL/commit/74a3e4f963a77ba43f2a2e2abe02bbeea22eba09
